src/h5web/visualizations/line/DataCurve.tsx

```diff
@@ -19,45 +22,82 @@ function DataCurve(props: Props): ReactElement {
   const {
     abscissas,
     ordinates,
+    errors,
+    showErrors,
     color = DEFAULT_COLOR,
     curveType = CurveType.LineOnly,
   } = props;
 
   const { camera } = useThree();
   const { abscissaScale, ordinateScale } = useCanvasScales();
 
-  const dataGeometry = useMemo(() => {
-    const points = ordinates.map((val, index) => {
-      const ordinate = ordinateScale(val);
-      return new Vector3(
-        abscissaScale(abscissas[index]),
-        // This is to avoid a three.js warning when ordinateScale(val) is Infinity
-        Number.isFinite(ordinate) ? ordinate : 0,
-        // Move NaN/Infinity out of the camera FOV (negative val for logScale).
-        // This allows to have only curve segments for the positive values
-        Number.isFinite(ordinate) ? 0 : camera.far
-      );
+  const points = useMemo(() => {
+    const dataPoints: Vector3[] = [];
+    const errorBarSegments: Vector3[] = [];
+    const errorCapPoints: Vector3[] = [];
```

Comment:
I was about to suggest initializing the arrays with `new Array(ordinates.length)`, and then assigning values instead of pushing them, but apparently the performance gain is highly debatable.

Reply:
Also, only `dataPoints` is ensured to have the same length as `ordinates`.

Reply:
Oh right, I must have been thinking this when looking at the previous version, when you were adding `undefined` values.